src/views/TypingGame.vue

```diff
@@ -26,210 +26,183 @@
-      "future",     // e.g., "will run"
-      "presentContinuous", // e.g., "running"
-      "pastContinuous",    // e.g., "was running"
-      "presentPerfect",    // e.g., "have run"
-      "pastPerfect",       // e.g., "had run"
-      "futureContinuous",  // e.g., "will be running"
-      "futurePerfect",     // e.g., "will have run"
-      "conditional"        // e.g., "would run"
-    ];
-    const tensePostfixes = [
-      "ing",        // Present Continuous, Gerunds
-      "ed",         // Past Simple
-      "s",          // Present Simple (third person singular)
-      "en",         // Past Participle (e.g., "written")
-      "will",       // Future Simple
-      "was/were",   // Past Continuous
-      "have/has",   // Present Perfect
-      "had",        // Past Perfect
-      "be",         // Future Continuous
-      "would"       // Conditional
-    ];
+    function appendWord() {
+      let firstWord = "";
+      let i = Math.floor(Math.random() * sentence.length);
+      let text = "";
+      for (let n = 0; n < sentence[i].length; n++) {
+        if (n === 0) {
+          firstWord = getWord(sentence[i][n]);
+          firstWord = firstWord[0].toUpperCase() + firstWord.slice(1)
+          text += firstWord;
+        } else {
+          console.log(sentence[i][n])
+          text += space.concat(getWord(sentence[i][n]));
+        }
+      }
+      text += period;
+      return text;
+    }
 
-    let chooseTense = Math.floor(Math.random() * (tenseWords.length - 0) + 0);
-    const wordIndexMax = 19;
-    const wordIndexMin = 0;
-    const multiplyer = (wordIndexMax - wordIndexMin) + wordIndexMin;
-    const space = " ";
-    const period = ".";
-    console.log(Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin))
-    const texts = [
-        nouns[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + verbs[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + adverbs[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + period,
-        nouns[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + verbs[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + adjectives[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + nouns[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + period,
-        adverbs[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + nouns[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + verbs[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + space + items[Math.floor(Math.random() * (wordIndexMax - wordIndexMin) + wordIndexMin)] + period,
-      ];
+    function getWord(typeIndex) {
+      return present[typeIndex][Math.floor(Math.random() * (present[typeIndex].length - 0) + 0)];
```

Comment:
Please explain this

Reply:
We have a 2D array of "present" tense words. The 2D array has multiple types of words and each type has an array of words to choose from. The typeIndex is the index of the type of word such as "subject", "verb" etc, and the random index is used to choose a random word from the array of words of the respective type. All together, this function returns a random word of a specific type.